### PR TITLE
[luci] Introduce ReplaceSubWithAddPass

### PR DESCRIPTION
**Pull request by llFreetimell**
Parent Issue : #7081 
Draft : #7082 

This commit introduces `ReplaceSubWithAddPass` in `luci`.

ONE-DCO-1.0-Signed-off-by: Seok NamKoong <seok9311@naver.com>

### CODE REVIEW

File: compiler/luci/pass/src/ReplaceSubWithAddPass.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Service/Nodes/CircleConst.h>
+
+namespace
+{
+
+bool replace_sub_with_const_rhs(luci::CircleSub *sub)
+{
+  auto const_rhs = dynamic_cast<luci::CircleConst *>(sub->y());
```
Comment (mhs4670go) [inline compiler/luci/pass/src/ReplaceSubWithAddPass.cpp:27]: Only y position is supported?

Comment (llFreetimell) [reply compiler/luci/pass/src/ReplaceSubWithAddPass.cpp:27]: For now, yes :)
Only `CircleSub(x, y) whose y is constant` supported

Comment (seanshpark) [reply compiler/luci/pass/src/ReplaceSubWithAddPass.cpp:27]: X - Y = X + (-Y), where Y is Const
but when X is Const,  X - Y != (-X) + Y
Am I right?

Comment (mhs4670go) [reply compiler/luci/pass/src/ReplaceSubWithAddPass.cpp:27]: It really is :)

**Review by mhs4670go — APPROVED** [review]
LGTM

File: compiler/luci/pass/src/ReplaceSubWithAddPass.cpp (hunk 1 above)
Comment (llFreetimell) [reply compiler/luci/pass/src/ReplaceSubWithAddPass.cpp:27]: Hmm... the equation `X - Y != (-X) + Y` is always correct regardless of "X is const?" so the answer is yes...

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**